google/cloud/spanner/client.cc

```diff
@@ -176,14 +177,25 @@ StatusOr<CommitResult> Client::Commit(
         return status;
       }
     }
-    // A transient failure (i.e., kAborted), so consider rerunning.
+    // A transient failure (e.g., kAborted), so consider rerunning.
     if (!rerun_policy->OnFailure(status)) {
       return status;  // reruns exhausted
     }
+    if (internal::IsSessionNotFound(status)) {
+      // Marks the session bad and creates a new Transaction for the next loop.
+      internal::Visit(txn, [](internal::SessionHolder& s,
+                              google::spanner::v1::TransactionSelector const&,
+                              std::int64_t) {
+        s->set_bad();
```

Comment:
Actually, it was nullptr in the first draft of my test, which is what caused the segv I mentioned yesterday. However, I don't *think* that it can be nullptr in real code.

That said, I was tempted to -- and would still be happy to -- change this to `if (s) s->set_bad()` to guard against this case. Preference?